config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/environment/EnvironmentConfigurationUtils.java

```diff
@@ -36,40 +46,91 @@
 
     private final JsonTools jsonTools;
     private final EntityLoaderRegistry entityLoaderRegistry;
+    private static final Map<String, Pair<String,String>> ENTITY_FILE_MAP = unmodifiableMap(createEntityFileMap());
 
     @Inject
     EnvironmentConfigurationUtils(JsonTools jsonTools, EntityLoaderRegistry entityLoaderRegistry) {
         this.jsonTools = jsonTools;
         this.entityLoaderRegistry = entityLoaderRegistry;
     }
 
-    @SuppressWarnings("unchecked")
-    public Map<String, String> parseEnvironmentValues(Map providedEnvironmentValues) {
-        final Map envConfig = ofNullable(providedEnvironmentValues).orElseThrow(() -> new MissingEnvironmentException("Environment configuration was not specified into gradle configuration file."));
-        final Map<String, String> environmentValues = new HashMap<>();
-        envConfig.entrySet().forEach((Consumer<Entry>) e -> environmentValues.put(PREFIX_ENV + e.getKey().toString(), getEnvValue(e.getKey().toString(), e.getValue())));
-
-        return unmodifiableMap(environmentValues);
+    /**
+     * Creates the Map of entity names with their environment config file names.
+     *
+     * @return entity names with config file names.
+     */
+    private static Map<String, Pair<String,String>> createEntityFileMap() {
+        Map<String, Pair<String,String>> entityFileMap = new HashMap<>();
+        entityFileMap.put(EntityTypes.ID_PROVIDER_CONFIG_TYPE, ImmutablePair.of("IDENTITY_PROVIDER", "identity-providers.yml"));
```

Comment:
you can get all the entity types with annotation EnvironmentType

Reply:
names are mismatching at @EnvironmentType, ex: CLUSTER_PROPERTY is referred as PROPERTY

Reply:
we can add condition for CLLUSTER PROPERTY, otherwise for every new environment type we should update this list

Reply:
EntityUtils.java has all utilities for getting gatewayentity infor

Reply:
done